RuntimeDetour/IDetourRuntimePlatform.cs

```diff
@@ -16,5 +16,11 @@ interface IDetourRuntimePlatform {
         void Pin(MethodBase method);
         void Unpin(MethodBase method);
         MethodBase GetDetourTarget(MethodBase from, MethodBase to);
+        event OnMethodCompiledEvent OnMethodCompiled;
```

Comment:
I'll add a property indicating whether or not it will be called. I don't think it would be too terribly difficult to hook Mono's JIT, if it ever becomes necessary (like for implementing sane generic hooks), but until then it shouldn't be an error to subscribe to it if the platform doesn't support it IMO.